test-dts/createRouter.test-d.ts

```diff
@@ -14,5 +20,12 @@ const router = createRouter({
   scrollBehavior(to, from, savedPosition) {},
 })
 
+export const loggedInGuard: NavigationGuard = (
```

Comment:
isn't this type redundant? I think the type arguments make more sense with a plain function:


```js
function otherGuard(to: RouteLocationNormalized, from: RouteLocationNormalized, next: NavigationGuardCallback) {}
```

I also think we should rename it to `NavigationGuardNext`, don't do it in the PR to keep it simple but what do you think?

Reply:
I think I agree that `NavigationGuard` is not super necessary. `NavigationGuardNext` is probably better indeed 👍 And `RouteLocationNormalized` could be simplified to `RouteLocation`?

Reply:
I want to but there are a few more types that represent different kinds of route locations. I think I need to do some refactoring when it comes to naming and would definitely welcome some feedback